core/src/main/scala/dagr/core/execsystem/GraphNode.scala

```diff
@@ -37,11 +39,10 @@ import scala.collection.mutable.ListBuffer
   *                         is used to track submission, start, and end dates for the parent node.  Typically parent nodes
   *                         are [[dagr.core.tasksystem.Pipeline]]s.
   */
-class GraphNode(val taskId: BigInt,
-                var task: Task,
+class GraphNode(var task: Task,
                 predecessorNodes: Traversable[GraphNode] = Nil,
                 var state: GraphNodeState.Value = GraphNodeState.PREDECESSORS_AND_UNEXPANDED,
-                val parent: Option[GraphNode] = None) {
+                val parent: Option[GraphNode] = None) extends BaseGraphNode {
```

Comment:
This is my first time reading some of the graph code, so I'm going to ask some basic questions.  I see this constructor takes both `predecessorNodes` and an optional `parent.  Is it the case that predecessors are nodes which must be succeeded before this node executes, and parent is the node that generated this node during expansion?  If that's right, I might suggest `enclosingNode` instead of parent to make the distinction clearer.